src/librbd/cache/pwl/rwl/WriteLog.cc

```diff
@@ -439,6 +439,19 @@ void WriteLog<I>::load_existing_entries(DeferredContexts &later) {
   this->update_sync_points(missing_sync_points, sync_point_entries, later, MIN_WRITE_ALLOC_SIZE);
 }
 
+template <typename I>
+void WriteLog<I>::calc_allocated_and_cached_data(std::shared_ptr<pwl::GenericLogEntry> log_entry, uint32_t alloc_size) {
+  auto gen_write_entry = static_pointer_cast<GenericWriteLogEntry>(log_entry);
+  if (log_entry->is_write_entry()) {
```

Comment:
Could you explain why this check ` if (log_entry->write_bytes() == log_entry->bytes_dirty())` is removed? That check will differentiate `writesame` from a `write` request. In `writesame`, we don't allocate all of the `write_bytes()` length. I think we need to differentiate when calculating `m_bytes_allocated`